qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -43,19 +55,23 @@ def __init__(self, data: Mapping[str, np.ndarray | Number], register_length: int
             if isinstance(value, Number):
                 value = np.asarray(value)
 
-            if len(value.shape) != len(key):
+            if validate and len(value.shape) != len(key):
                 raise ValueError(
                     f"Data key {key} of length {len(key)} does not match "
                     f"data value matrix of dimensions {value.shape}"
                 )
 
             dims = set(value.shape)
 
-            if len(dims) > 1:
+            if validate and len(dims) > 1:
```

Comment:
We want to be able to disable this check, in order to have the ability to do e.g. active-space transformations which map from `n` AOs to `m` MOs

Reply:
Yes that's what is required for the same objects to be usable in a context like basis transforms to a reduced dimension.
I don't think it complicates things much. `numpy` will automatically raise errors when a user is trying to combine mismatching objects. But yes, it does require more cognitive awareness of the user who is doing such operations.

Reply:
Reading your comment again, let me revise my previous answer:
I do not want to store matrices of different size, I just would like to support non-square matrices. For example:
```python
ao2mo = np.zeros((4, 2))
ao2mo[...] = ...
# this 4x2 matrix, maps from 4 AOs to 2 MOs
tensor = PolynomialTensor({"+-": ao2mo}, validate=False)
```